src/Event/Listener/ContentFillListener.php

```diff
@@ -72,4 +85,74 @@ private function guesstimateAuthor(): User
     {
         return $this->users->getFirstAdminUser();
     }
+
+    private function guaranteeUniqueSLug(Content $content): void
+    {
+        $slug = $content->getSlug();
+        $slugField = $content->getField('slug') ?? null;
+
+        $fields = $this->fieldRepository->findAllBySlug($slug);
+
+        if (! $fields) {
+            // No slug field with that slug exists. We're done here.
+            return;
+        }
+
+        $safe = true;
+
+        // Clone fields for each locale
+        $tempFields = [];
+        /** @var Field $field */
+        foreach ($fields as $field) {
+            foreach ($field->getTranslations() as $translation) {
+                $tempFields[] = clone $field->setLocale($translation->getLocale());
+            }
+        }
+        $fields = $tempFields;
+
+        /** @var Field $field */
+        foreach ($fields as $field) {
+            // If the contenttype slugs are different, we're safe.
+            if ($field->getContent()->getContentTypeSlug() !== $content->getContentTypeSlug()) {
+                continue;
+            }
+
+            // If the locales are different, we're safe.
+            if ($slugField && $slugField->getLocale() !== $field->getLocale()) {
+                continue;
+            }
+
+            // If the content and locale is the same, we're safe.
+            if ($content === $field->getContent()) {
+                continue;
+            }
+
+            $safe = false;
+
+            break;
+        }
+
+        // If we're not safe, use recursion to find safe slug
+        if (! $safe && $slugField) {
+            $newSlug = $this->getSafeSlug($slug);
+            $slugField->setValue($newSlug);
+            $this->guaranteeUniqueSLug($content);
+        }
+    }
+
+    private function getSafeSlug(string $slug): string
+    {
+        $separator = '-';
+        // If it already ends with -{number}, increase it!
+        $dashNumber = '/' . $separator . "(\d)$/";
```

Comment:
I think this matches only one number.. suggestion: 

```suggestion
        $dashNumber = '/' . $separator . "(\d+)$/";
```

Reply:
Yess!